components/ui/Button.tsx

```diff
@@ -1,76 +1,185 @@
-    textStyles?: TextStyle
+  variant?: ButtonVariant;
+  textSize?: SizeVariants;
+  title?: string;
+  active?: boolean;
+  containerStyles?: ViewStyle;
+  textStyles?: TextStyle;
+  disabled?: boolean;
+  fullWidth?: boolean;
+  children?: React.ReactNode;
 }
-export function CustomButton(
-    {
-        variant = 'primary',
-        onPress,
-        textSize,
-        textStyles = {},
-        containerStyles = {},
-        active,
-        title,
-        ...rest
-    }: ButtonProps): JSX.Element {
-
-    const variantStyles = {
-        primary: {
-            container: styles.primary,
-            text: styles.primaryVariantText,
+export function CustomButton({
+  variant = "default",
+  onPress,
+  textSize,
+  textStyles = {},
+  containerStyles = {},
+  active,
+  disabled,
+  children,
+  fullWidth,
+  title,
+  ...rest
+}: ButtonProps): JSX.Element {
+  const getVariantStyles = (
+    variant: string
+  ): { container: ViewStyle; text?: TextStyle } => {
+    if (variant === "primary") {
+      return {
+        container: styles.primary,
+        text: styles.primaryText,
+      };
+    }
+    if (variant === "secondary") {
+      return {
+        container: {
+          ...(fullWidth && styles.secondaryFullWidth),
+          ...styles.secondary,
         },
-        secondary: {
-            container: [styles.secondary, active && styles.secondaryActive],
-            text: [styles.secondaryVariantText, active && styles.secondaryActiveVariantText],
-        }
+        text: styles.secondaryText,
+      };
     }
+    if (variant === "outline") {
+      return {
+        container: styles.outline,
+        text: styles.outlineText,
+      };
+    }
+    if (variant === "destructive") {
+      return {
+        container: styles.destructive,
+        text: styles.descructiveText,
+      };
+    }
+    if (variant === "ghost") {
+      return {
+        container: styles.ghost,
+      };
+    }
+    if (variant === "default") {
+      return {
+        container: styles.default,
+        text: styles.secondaryText,
+      };
+    }
+    return {
+      container: {},
+      text: {},
+    };
+  };
+  const { container, text } = getVariantStyles(variant);
 
-    const { container, text } = variantStyles[variant] || variantStyles.primary;
-
-    return (
-        <TouchableOpacity onPress={onPress} style={[container, containerStyles]}  {...rest} >
-            <CustomText size={textSize} style={[text, textStyles]}>{title}</CustomText>
-        </TouchableOpacity>
-    )
+  return (
+    <TouchableOpacity
+      onPress={onPress}
+      style={[
+        styles.buttonContainer,
+        container,
+        containerStyles,
+        disabled && styles.disabled,
+      ]}
+      {...rest}
+    >
+      {variant === "ghost" && children ? (
+        children
+      ) : (
+        <CustomText
+          size={textSize}
+          style={[text, textStyles, styles.buttonText]}
+        >
+          {title}
+        </CustomText>
+      )}
```

Comment:
we should allow buttons with all variants to have children, for instance this one will probably use children and it's not a ghost variant
![image](https://github.com/user-attachments/assets/d38c3a28-f2d6-4d3c-818a-45850d09dbee)